### PR TITLE
chore: readme for testing/building client with on posix

### PR DESCRIPTION
**Pull request by danielskinstad**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,40 @@ Since the project is under active development, we recommend watching the reposit
...
+- cmake
+- libcurl4-openssl-dev
+- libcjson-dev
+- libmbedtls-dev
+
+### Building the Client
+1. Configure the build:
+   ```
+   cmake -B build tests -DCONFIG_MENDER_PLATFORM_FLASH_TYPE="posix" -DCONFIG_MENDER_PLATFORM_LOG_TYPE="posix" -DCONFIG_MENDER_PLATFORM_NET_TYPE="generic/curl" -DCONFIG_MENDER_PLATFORM_SCHEDULER_TYPE="posix" -DCONFIG_MENDER_PLATFORM_STORAGE_TYPE="posix" -DCONFIG_MENDER_PLATFORM_TLS_TYPE="generic/mbedtls" -DCONFIG_MENDER_CLIENT_ADD_ON_INVENTORY=ON -DCONFIG_MENDER_CLIENT_ADD_ON_CONFIGURE=OFF -DCONFIG_MENDER_CLIENT_CONFIGURE_STORAGE=OFF -DCONFIG_MENDER_CLIENT_ADD_ON_TROUBLESHOOT=OFF -DCONFIG_MENDER_FULL_PARSE_ARTIFACT=ON -DCONFIG_MENDER_PROVIDES_DEPENDS=ON
+   ```
```
Comment (larsewi) [inline README.md:47]: Remove leading spaces

File: README.md [hunk 2]
```diff
@@ -32,6 +32,40 @@ Since the project is under active development, we recommend watching the reposit
...
+
+2. Build the client:
+   ```
+   cmake --build build --parallel $(nproc --all)
+   ```
```
Comment (larsewi) [inline README.md:52]: Same here

File: README.md [hunk 3]
```diff
@@ -32,6 +32,40 @@ Since the project is under active development, we recommend watching the reposit
...
+
+### Running the Client
+You can now run and connect the client to e.g. hosted Mender:
+```
+./build/mender-mcu-client.elf --mac_address=$MAC_ADDRESS --device_type=$DEVICE_TYPE --tenant_token=$TENANT_TOKEN --artifact_name=$ARTIFACT_NAME
```
Comment (larsewi) [inline README.md:57]: You can maybe add some exports above. E.g.:
```
export MAC_ADDRESS=<YOUR MAC ADDRESS>
...
```
For `MAC_ADDRESS` you can mention that it does not really matter what you put there. For `TENANT_TOKEN` where you find this. And so on...

File: README.md [hunk 4]
```diff
@@ -32,6 +32,40 @@ Since the project is under active development, we recommend watching the reposit
...
+
+### Creating an Artifact
+Create an artifact (remember to disable compression):
+```
+./mender-artifact write rootfs-image --compression none --artifact-name <artifact_name> --device-type <device_type> --file <file_name>
+```
+The `device_type` in the artifact has to match the `device_type` used when running the client.
+
+### Deployment
+After creating and uploading the artifact to the server, you should be able to deploy it to the device.
```
Comment (larsewi) [inline README.md:68]: What are the steps to do this? How do you accept a device to hosted mender? Questions like these does not hurt to answer here. Also some screenshots might be nice. In general, expect that the readers are not as smart as you are ;)

File: README.md [hunk 5]
```diff
@@ -32,6 +32,40 @@ Since the project is under active development, we recommend watching the reposit
...
+### Dependencies
+- cmake
+- libcurl4-openssl-dev
+- libcjson-dev
+- libmbedtls-dev
+
+### Building the Client
+1. Configure the build:
+   ```
+   cmake -B build tests -DCONFIG_MENDER_PLATFORM_FLASH_TYPE="posix" -DCONFIG_MENDER_PLATFORM_LOG_TYPE="posix" -DCONFIG_MENDER_PLATFORM_NET_TYPE="generic/curl" -DCONFIG_MENDER_PLATFORM_SCHEDULER_TYPE="posix" -DCONFIG_MENDER_PLATFORM_STORAGE_TYPE="posix" -DCONFIG_MENDER_PLATFORM_TLS_TYPE="generic/mbedtls" -DCONFIG_MENDER_CLIENT_ADD_ON_INVENTORY=ON -DCONFIG_MENDER_CLIENT_ADD_ON_CONFIGURE=OFF -DCONFIG_MENDER_CLIENT_CONFIGURE_STORAGE=OFF -DCONFIG_MENDER_CLIENT_ADD_ON_TROUBLESHOOT=OFF -DCONFIG_MENDER_FULL_PARSE_ARTIFACT=ON -DCONFIG_MENDER_PROVIDES_DEPENDS=ON
```
Comment (lluiscampos) [inline README.md:46]: Specifying here the defines will soon get out of date. Suggestion: create a one liner wrapper script for this and call it both from here and from the CI job :bulb:

File: README.md (hunk 4 above)
Comment (lluiscampos) [reply README.md:68]: @larsewi We can safely assume that any user reading this knows Mender. On our final README we will have links to docs.mender.io and there there is a tutorial on how to do your first deployment, etc.

Specially screenshots will soon get out of date, we should have them only in docs-mender.io

File: README.md [hunk 6]
```diff
@@ -32,6 +32,40 @@ Since the project is under active development, we recommend watching the reposit
 regularly for updates. Detailed documentation and usage instructions will be provided as the project
 progresses.
 
+## Testing the Mender MCU Client (POSIX)
+
+### Dependencies
+- cmake
+- libcurl4-openssl-dev
+- libcjson-dev
+- libmbedtls-dev
```
Comment (vpodzime) [inline README.md:41]: I'd prefer these being generic things like `CMake`, `libcurl`, `cJSON` and `mbedTLS` and not Ubuntu/Debian-specific packages.

Comment (larsewi) [reply README.md:41]: I guess you can do both. List up what you need, and provide an example with an apt command for ubuntu

File: README.md [hunk 7]
```diff
@@ -32,6 +32,52 @@ Since the project is under active development, we recommend watching the reposit
...
+- mbedTLS
+
+Example for Ubuntu/Debian:
+```
+apt install cmake libcurl4-openssl-dev libmbedtls-dev
+```
+### Building the Client
+1. Configure the build:
+```
+./build_posix.sh
```
Comment (larsewi) [inline README.md:50]: Should be `./configure_posix.sh` ?

File: README.md [hunk 8]
```diff
@@ -32,6 +32,52 @@ Since the project is under active development, we recommend watching the reposit
...
+- CMake
+- libcurl
+- cJSON
+- mbedTLS
+
+Example for Ubuntu/Debian:
+```
+apt install cmake libcurl4-openssl-dev libmbedtls-dev
+```
+### Building the Client
```
Comment (larsewi) [inline README.md:47]:
```suggestion

### Building the Client
```

**Review by larsewi — APPROVED** [review]
Otherwise it looks good. Thanks for sharing this 🚀

File: configure_posix.sh [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+-DCONFIG_MENDER_PLATFORM_NET_TYPE="generic/curl" \
+-DCONFIG_MENDER_PLATFORM_SCHEDULER_TYPE="posix" \
+-DCONFIG_MENDER_PLATFORM_STORAGE_TYPE="posix" \
+-DCONFIG_MENDER_PLATFORM_TLS_TYPE="generic/mbedtls" \
+-DCONFIG_MENDER_CLIENT_ADD_ON_INVENTORY=ON \
+-DCONFIG_MENDER_CLIENT_ADD_ON_CONFIGURE=OFF \
+-DCONFIG_MENDER_CLIENT_CONFIGURE_STORAGE=OFF \
+-DCONFIG_MENDER_CLIENT_ADD_ON_TROUBLESHOOT=OFF \
+-DCONFIG_MENDER_FULL_PARSE_ARTIFACT=ON \
+-DCONFIG_MENDER_PROVIDES_DEPENDS=ON
```
Comment (larsewi) [inline configure_posix.sh:15]: Can you please rebase and add this option? :) 
```suggestion
-DCONFIG_MENDER_PROVIDES_DEPENDS=ON \
-DCONFIG_MENDER_ALL_WARNINGS_AS_ERRORS=ON
```

**Review by vpodzime — APPROVED** [review]
The second commit is no longer adding a script so the commit message needs a bit of rewording. Hats off otherwise and THANKS!

**Pull request merged**